pkgs/applications/window-managers/xborders/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, python3Packages, fetchFromGitHub, libwnck, gtk3, libnotify, wrapGAppsHook, gobject-introspection }:
+
+with python3Packages; buildPythonPackage rec {
+  pname = "xborders";
+  version = "3.4"; # in version.txt
+  src = fetchFromGitHub {
+    owner = "deter0";
+    repo = "xborder";
+    rev = "d840ed649e92b7ad399cdb1a30aeca7d09f2c08f";
+    sha256 = "sha256-Q2WWXekUL/OOr/OBqJOZbAHaWURbP+coBVp206uPSNY=";
+  };
+  buildInputs = [ libwnck gtk3 libnotify ];
+  nativeBuildInputs = [ wrapGAppsHook gobject-introspection ];
+  propagatedBuildInputs = [ pycairo requests pygobject3 ];
+  preBuild = ''
```

Comment:
I should be a `postPhase`.

Reply:
You mean `preBuild` should be `postPhase`? It doesn't build with `postPhase` because `setup.py` needs to be created before building, otherwise `buildPythonPackage` fails.

Reply:
Sorry, I meant `postPatch`. `preBuild` should be `postPatch` because, semantically, it is a patch - you are modifying the source files.

Also, you should include this `setup.py` file as a patch too - or at least directly as a file, not as a heredoc.